src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -10737,5 +10723,842 @@ struct S
+            Assert.Equal(NullableAnnotation.NotAnnotated, prop.BackingField.TypeWithAnnotations.NullableAnnotation);
+        }
+
+        [Fact]
+        public void Nullable_Resilient_AnnotationInMetadata()
+        {
+            // Inferred nullability is not used in metadata.
+            var source = """
+                #nullable enable
+
+                class C
+                {
+                    public string Prop
+                    {
+                        get => field ??= "a";
+                    }
+                }
+                """;
+
+            var comp0 = CreateCompilation(source);
+            comp0.VerifyEmitDiagnostics();
+            var sourceField = comp0.GetMember<SynthesizedBackingFieldSymbol>("C.<Prop>k__BackingField");
+            Assert.True(sourceField.InfersNullableAnnotation);
+            Assert.Equal(NullableAnnotation.NotAnnotated, sourceField.TypeWithAnnotations.NullableAnnotation);
+            Assert.Equal(NullableAnnotation.Annotated, sourceField.GetInferredNullableAnnotation());
+
+            var comp1 = CreateCompilation("", references: [comp0.EmitToImageReference()], options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            var metadataField = comp1.GetMember<FieldSymbol>("C.<Prop>k__BackingField");
+            Assert.Equal(NullableAnnotation.NotAnnotated, metadataField.TypeWithAnnotations.NullableAnnotation);
+        }
+
+        [Fact]
+        public void Nullable_DisallowNullField_NullableValueType()
+        {
+            var source = """
+                #nullable enable
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    public C()
+                    {
+                        Prop = null; // 1
+                        Prop = 0;
+                    }
+
+                    [field: DisallowNull]
+                    public int? Prop
+                    {
+                        get => field;
+                    }
+                }
+                """;
+
+            var comp0 = CreateCompilation([source, DisallowNullAttributeDefinition]);
+            comp0.VerifyEmitDiagnostics(
+                // (8,16): warning CS8607: A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]
+                //         Prop = null; // 1
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "null").WithLocation(8, 16));
+
+            var sourceField = comp0.GetMember<SynthesizedBackingFieldSymbol>("C.<Prop>k__BackingField");
+            Assert.False(sourceField.InfersNullableAnnotation);
+            Assert.Equal(NullableAnnotation.Annotated, sourceField.TypeWithAnnotations.NullableAnnotation);
+            Assert.Equal("System.Int32?", sourceField.TypeWithAnnotations.ToTestDisplayString());
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing a `static` property and associated constructor.

Reply:
Done